testsuites/CBLTester/CBL_Functional_tests/cbl_log_decoder_tool/test_cbl_log_decoder.py

```diff
@@ -0,0 +1,183 @@
+
+    log_dir = "resources/data/cbl_logs"
+    cbl_platforms = ["iOS", "xamarin-android", "Net-core", "uwp"]
+    versions = ["2.5.0"]
+
+    for platform in cbl_platforms:
+        for version in versions:
+            cbl_log_dir = "{}/{}/{}".format(log_dir, version, platform)
+            cbl_logs = os.listdir(cbl_log_dir)
+            extracted_cbllog_directory_name = download_cbl_log(cbl_log_decoder_platform, liteserv_version, cbl_log_decoder_build)
+            for log in cbl_logs:
+                file = "{}/{}".format(cbl_log_dir, log)
+                assert is_binary(file), "{} file is not binary ".format(file)
+                decode_logs(extracted_cbllog_directory_name, file)
+
+
+@pytest.mark.listener
+@pytest.mark.database
+def test_cbl_decoder_with_current_logs(params_from_base_test_setup):
+    """
+    @summary:
+    1.Get the cbl log decoder tool based on the build number provided
+    2.Run some basic flow on cbl like create bulk docs
+    3. Pull logs from cbl client to local machine to /tmp/test directory
+    4. Verify logs are binary
+    5. Use the decoder tool and decode the logs and verify logs are not binary after decoding
+    """
+    base_url = params_from_base_test_setup["base_url"]
+    liteserv_version = params_from_base_test_setup["liteserv_version"]
+    liteserv_platform = params_from_base_test_setup["liteserv_platform"]
+    cbl_log_decoder_platform = params_from_base_test_setup["cbl_log_decoder_platform"]
+    cbl_log_decoder_build = params_from_base_test_setup["cbl_log_decoder_build"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+
+    log_obj = FileLogging(base_url)
+    num_of_docs = 10
+    channels_sg = ["ABC"]
+    log_level = "verbose"
+    plain_text = False
+    if liteserv_platform == "net-msft" or liteserv_platform == "net-uwp":
+        cbl_log_dir = "c:/test"
+    else:
+        cbl_log_dir = "/tmp/test"
+    # Clean up test directory to remove stale logs
+    if os.path.exists(cbl_log_dir):
+        shutil.rmtree(cbl_log_dir)
+    os.makedirs(cbl_log_dir)
+    log_obj.configure(log_level=log_level, plain_text=plain_text, directory=cbl_log_dir)
+    log_directory = log_obj.get_directory()
+    log_info("logs are enabled at: {}".format(log_directory))
+    if liteserv_version < "2.5.0":
+        pytest.skip('This test cannot run with CBL version below 2.5.0')
+
+    db.create_bulk_docs(num_of_docs, id_prefix="cbl-decoder", db=cbl_db, channels=channels_sg)
+
+    get_logs(liteserv_platform, log_directory)
+    extracted_cbllog_directory_name = download_cbl_log(cbl_log_decoder_platform, liteserv_version, cbl_log_decoder_build)
+    cbl_logs = os.listdir(cbl_log_dir)
+    for log in cbl_logs:
+        file = "{}/{}".format(cbl_log_dir, log)
+        assert is_binary(file), "{} file is not binary ".format(file)
+        decode_logs(extracted_cbllog_directory_name, file)
+
+
+def get_logs(liteserv_platform, log_directory):
```

Comment:
where this method handles iOS/Xamarin-iOS scenario?

Reply:
It is good question. It is not required for iOS and xamarin-ios . When we give custom logging directory, it stores in local path for iOS and xamarin-iOS. Only for Android and windows , we need to pull logs